processor/tailsampling/processor.go

```diff
@@ -69,40 +72,41 @@ const (
 	sourceFormat = "tail-sampling"
 )
 
-var _ consumer.TraceConsumer = (*tailSamplingSpanProcessor)(nil)
+var _ processor.TraceProcessor = (*tailSamplingSpanProcessor)(nil)
 
-// NewTailSamplingSpanProcessor creates a TailSamplingSpanProcessor with the given policies.
-// It will keep maxNumTraces on memory and will attempt to wait until decisionWait before evaluating if
-// a trace should be sampled or not. Providing expectedNewTracesPerSec helps with allocating data structures
-// with closer to actual usage size.
-func NewTailSamplingSpanProcessor(
-	policies []*Policy,
-	maxNumTraces, expectedNewTracesPerSec uint64,
-	decisionWait time.Duration,
-	logger *zap.Logger) (consumer.TraceConsumer, error) {
+// NewTraceProcessor returns a processor.TraceProcessor that will perform tail sampling according to the given
+// configuration.
+func NewTraceProcessor(logger *zap.Logger, nextConsumer consumer.TraceConsumer, cfg builder.TailBasedCfg) (processor.TraceProcessor, error) {
```

Comment:
Is there a good reason that `TailBasedCfg` is in `builder` package? If not then perhaps it should be in this package?
Actually, I am not sure `sampling_builder.go` should exist in `builder` package at all, but that's probably a topic for a separate cleanup PR.

Reply:
I think that config is inherited from opencensus-service where sampling is configured through builder. I'll move it to processor/tailsampling and clean up the old configs in the next PR.